### ISSUE TITLE
[terra-collapsible-menu-view] Enhancement to allow always collapsed menu items

### ISSUE THREAD

**Issue by EPAMMarcell** [cerner/terra-framework#1296]
# Feature Request
We would like to request an update for **terra-collapsible-menu-view** with a new feature:
- optional parameter (an array) of items, which should populate the submenu (be always in the submenu) instead of the default menu-items. Even if there is enough space to show them.
This should make the three dot (thin hamburger) button always shown, since there are (at least one) menu item inside the submenu dropdown.

## Description
Business requirements for physician documentation requires these abilities for the toolbar of autotext phrases (with favorites).
Jira: PHYSDOCSVC-9823
as we see now the CollapsibleMenuView doesn't support these functions.
We would like to achieve the following:
* AC#1 Make Collapsible menu view visible when at least one menu item is added
* AC#2 Reorder option appears always in the drop-down (as a default menuitem)
Basically: we want to introduce a new functionality in this component: an option to be able to always show the 'three dot / thin hamburger' menu button at the end whenever we have at least one item in the whole menu item list. Also, we would want to populate the submenu with at least one menuitem (reorder).

## Additional Context / Screenshots
Related miro board: Autotext
The UX for the collapsible menu view: attached to jira

## @ Mentions
@ryanthemanuel

**Comment by EPAMMarcell** [comment]
@ryanthemanuel 

### API spec update
(more of a prop update)
introduce new prop in **CollapsibleMenuViewItem** component: isCollapsed
**isCollapsed** (bool): Puts the menu item under the collapsed (more) menu. Note: More button will be always shown if at least one item is marked as collapsed.
alternative name ideas: isAlwaysCollapsed, isAlwaysHidden (because in the CollapsibleMenuView visible and hidden children is used as terms for the menu items which are always shown and which are put under the more button).
If you have a favorit or something else would be considered more accurate and more domain specific we can update it the way you would like it.
### POC demo
![CollapsibleMenuViewAlwaysCollapsedPOCDemo](https://user-images.githubusercontent.com/60744517/100120078-d7b6cf00-2e77-11eb-9922-15192f836e58.gif)

**Comment by dkasper-was-taken** [comment]
@EPAMMarcell We will want to avoid setting an isCollapsed per item as it could result in item reordering on resize or confusion from our users if we prevent it from reordering. 
The desired behavior would be all items either above a specific index or all items within a given bucket as hidden. The bucket would likely be more explicit for our consumers, so that would be the ideal behavior.

I put together a prototype using the bucket approach.
https://github.com/cerner/terra-framework/compare/alwayshidden-prototype

The prop is an array of items that is concatted onto the existing hidden items. Within the resize logic we calculate assuming the more button always if the hidden items bucket is present.

**Comment by EPAMMarcell** [comment]
PR Merged.
